tsl/src/compression/compression.c

```diff
@@ -2689,65 +2993,143 @@ decompress_batches(RowDecompressor *decompressor, ScanKeyData *scankeys, int num
 						  decompressor->compressed_datums,
 						  decompressor->compressed_is_nulls);
 
-		TM_FailureData tmfd;
-		TM_Result result;
-		result = table_tuple_delete(decompressor->in_rel,
-									&compressed_tuple->t_self,
-									decompressor->mycid,
-									snapshot,
-									InvalidSnapshot,
-									true,
-									&tmfd,
-									false);
+		result = delete_compressed_tuple(decompressor, compressed_tuple);
+		if (result != TM_Ok)
+		{
+			table_endscan(heapScan);
+			report_error(result);
+		}
+		row_decompressor_decompress_row(decompressor, NULL);
+		*chunk_status_changed = true;
+	}
+	if (scankeys)
+		pfree(scankeys);
+	table_endscan(heapScan);
+	return true;
+}
 
-		switch (result)
+/*
+ * This method will build scan keys required to do index
+ * scans on compressed chunks.
+ */
+static List *
+build_index_scankeys(Relation in_rel, Relation index_rel, List *predicates, List **num_scankeys,
+					 EState *estate)
+{
+	List *scankeys = NIL;
+	IndexScan *node = makeNode(IndexScan);
+	Plan *plan = &node->scan.plan;
+
+	plan->targetlist = NULL;
+	plan->qual = predicates;
+	plan->lefttree = NULL;
+	plan->righttree = NULL;
+	node->scan.scanrelid = in_rel->rd_id;
+	node->indexid = index_rel->rd_id;
+	node->indexqual = predicates;
+	node->indexqualorig = NULL;
+	node->indexorderby = NULL;
+	node->indexorderbyorig = NULL;
+	node->indexorderbyops = NULL;
+	node->indexorderdir = ForwardScanDirection;
```

Comment:
makeNode uses palloc0 so explicit NULL initialization is redundant
```suggestion
	plan->qual = predicates;
	node->scan.scanrelid = in_rel->rd_id;
	node->indexid = index_rel->rd_id;
	node->indexqual = predicates;
	node->indexorderdir = ForwardScanDirection;
```

Reply:
Fixed.